script.js

```diff
@@ -1,12 +1,247 @@
+  footer.appendChild(licensing);
+
+  // append the footer to the body
+  body.appendChild(footer);
+}
+
+//level 300-adding an episode selector 
+function episodeSelectField(episodes) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  const episodeSelectContainer = document.createElement("div");
+  episodeSelectContainer.classList.add("episode-select-container");
+
+  const episodeSelect = document.createElement("select");
+  episodeSelect.classList.add("episode-select");
+  episodeSelect.addEventListener("change", () => {
+    const episodeId = episodeSelect.value;
+    if (episodeId === "select-episode") {
+      makePageForEpisodes(allEpisodes);
+    } else {
+      const selectedEpisode = allEpisodes.find((episode) => episode.id == episodeId);
+      makePageForEpisodes([selectedEpisode]);
+    }
+  });
+
+  const selectOption = document.createElement("option");
+  selectOption.value = "select-episode";
+  selectOption.textContent = "Select Episode";
+  episodeSelect.appendChild(selectOption);
+
+  episodes.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = episode.id;
+    option.textContent = `${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    episodeSelect.appendChild(option);
+  });
+
+  episodeSelectContainer.appendChild(episodeSelect);
+  searchSelectDiv.appendChild(episodeSelectContainer);
 }
 
-function makePageForEpisodes(episodeList) {
+// Level 200 - Add search field for episodes
+function addSearchField() {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  const searchContainer = document.createElement("div");
+  searchContainer.classList.add("search-container");
+
+  const searchForm = document.createElement("form");
+  searchForm.classList.add("search-form");
+
+  const searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.type = "search";
+  searchInput.placeholder = "Search episodes";
+  searchInput.id = "search-input";
+
+  const searchLabel = document.createElement("label");
+  searchLabel.classList.add("search-label");
+  searchLabel.setAttribute("for", "search-input");
+
+  searchForm.appendChild(searchInput);
+  searchForm.appendChild(searchLabel);
+  searchContainer.appendChild(searchForm);
+  searchSelectDiv.appendChild(searchContainer);
+
+  // add event listener to the search input
+  searchInput.addEventListener("input", () => {
+    const query = searchInput.value.toLowerCase();
+    const filteredEpisodes = allEpisodes.filter((episode) => {
+      const title = `${episode.name} - S${episode.season
+        .toString()
+        .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+      const titleMatch = title.toLowerCase().includes(query);
+      const summaryMatch = episode.summary.toLowerCase().includes(query);
```

Comment:
your code should handle the case where the episode has no summary at all. For example, the show "World Series of Poker" breaks your search.

Reply:
thank you  , Jonathan 
I didn't thought about that , i will make sure to implement it